CLUE_Rock_Paper_Scissors/advanced/rps_crypto.py

```diff
@@ -0,0 +1,100 @@
+# MIT License
+
+# Copyright (c) 2020 Kevin J. Walters
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+import os
+import random
+
+from rps_crypto_chacha import ChaCha
+
+
+def bytesPad(text, size=8, pad=0):
+    """Convert a string to bytes and add pad value if necessary to make the length up to size.
+       """
+    text_as_bytes = text.encode("utf-8")
+    if len(text_as_bytes) >= size:
+        return text_as_bytes
+    else:
+        return text_as_bytes + bytes([pad] * (size - len(text_as_bytes)))
+
+
+def strUnpad(text_as_bytes, pad=0):
+    """Convert a bytes or bytearray
+       to a str removing trailing characters matching pad."""
+    if pad is not None:
+        end_ex = len(text_as_bytes)
+        while end_ex > 0 and text_as_bytes[end_ex - 1] == pad:
+            end_ex -= 1
+
+    # bytearray does not have decode method so need to convert here
+    return bytes(text_as_bytes[0:end_ex]).decode("utf-8")
```

Comment:
Could you use `rstrip()` here?

Reply:
Yes, this originated from `bytearray()` not having much to offer, all the cool functionality is in `bytes()`. But, I then needed `decode` too so I do end up going via `bytes()` and need to revisit that, thanks.

Reply:
Done